hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCacheTestSupport.java

```diff
@@ -529,6 +539,116 @@ public void run() throws Exception {
+        assertEquals("got the wrong expiration count", 0, stats.getExpirations());
+        assertEquals("we expect the same hits", statsBeforeEviction.getHits(), stats.getHits());
+        assertEquals("we expect the same misses", statsBeforeEviction.getMisses(), stats.getMisses());
+    }
+
+    protected void testNearCacheExpiration_withTTL(InMemoryFormat inMemoryFormat) {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat);
+        nearCacheConfig.setTimeToLiveSeconds(MAX_TTL_SECONDS);
+
+        testNearCacheExpiration(nearCacheConfig, MAX_TTL_SECONDS);
+    }
+
+    protected void testNearCacheExpiration_withMaxIdle(InMemoryFormat inMemoryFormat) {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat);
+        nearCacheConfig.setTimeToLiveSeconds(MAX_IDLE_SECONDS);
+
+        testNearCacheExpiration(nearCacheConfig, MAX_IDLE_SECONDS);
+    }
+
+    private void testNearCacheExpiration(NearCacheConfig nearCacheConfig, int expireSeconds) {
+        final int size = 147;
+
+        final NearCacheTestContext context = createNearCacheTest(DEFAULT_CACHE_NAME, nearCacheConfig);
+
+        for (int i = 0; i < size; i++) {
+            context.cache.put(i, "value-" + i);
+            context.cache.get(i);
+        }
+
+        assertEquals("we expected to have all map entries in the Near Cache", size,
+                getNearCacheStats(context.cache).getOwnedEntryCount());
+
+        final NearCacheStats statsBeforeExpiration = getNearCacheStats(context.cache);
+
+        sleepSeconds(expireSeconds + 1);
+
+        assertTrueEventually(new AssertTask() {
+            public void run() {
+                // map.get() triggers Near Cache eviction/expiration process,
+                // but we need to call this on every assert since the Near Cache has a cooldown for TTL cleanups
+                context.cache.get(0);
+
+                NearCacheStats stats = getNearCacheStats(context.cache);
+                assertEquals("we expect the same hits", statsBeforeExpiration.getHits(), stats.getHits());
+                assertEquals("we expect the same misses", statsBeforeExpiration.getMisses(), stats.getMisses());
+                assertEquals("we expect all entries beside the 'trigger entry' to be deleted from the Near Cache",
+                        1, stats.getOwnedEntryCount());
+                assertEquals("we did not expect any entries to be evicted from the Near Cache",
+                        0, stats.getEvictions());
+                assertTrue(format("we expect at least %d entries to be expired from the Near Cache", size),
+                        stats.getExpirations() >= size);
+            }
+        });
+    }
+
+    protected NearCacheStats getNearCacheStats(ICache cache) {
+        return cache.getLocalCacheStatistics().getNearCacheStatistics();
+    }
+
+    protected void waitForNearCacheEvictions(ICache cache, int evictionCount) {
+        long started = System.nanoTime();
```

Comment:
Can't you make use of assertTrueEventually? This prevents repeating the same logic.